code/Fluent.php

```diff
@@ -378,7 +378,7 @@ public static function domain_for_locale($locale)
      */
     public static function default_locale($domain = null)
     {
-        if ($domain === true) {
+        if (!Director::is_cli() && $domain === true) {
             $domain = strtolower($_SERVER['HTTP_HOST']);
```

Comment:
Instead of the is_cli() check, please use:

```php
$protocolAndHost = Director::protocolAndHost();
$protocol = Director::protocol();
$domain = substr($protocolAndHost, 0, strlen($protocol));
```

In ss 4.0 and above you can just use `$domain = Director::host()` but that's not yet implemented.
 
 `Director::protocolAndHost()` and remove the protocol.